src/envoy/mixer/http_filter.cc

```diff
@@ -77,36 +77,47 @@ const std::set<std::string> RequestHeaderExclusives = {
 // Set of headers excluded from response.headers attribute.
 const std::set<std::string> ResponseHeaderExclusives = {};
 
-std::shared_ptr<Auth::IssuerInfo> CreateIssuer(const JWT& jwt) {
-  std::vector<std::string> audiences;
+// Fill IssuerConfig from JWT proto
+void FillIssuerConfig(const JWT& jwt, Auth::IssuerConfig* issuer) {
   for (const auto& audience : jwt.audiences()) {
-    audiences.push_back(audience);
+    issuer->audiences.insert(audience);
   }
-  return std::make_shared<Auth::IssuerInfo>(
-      jwt.issuer(), jwt.jwks_uri(), jwt.jwks_uri_envoy_cluster(),
-      Auth::Pubkeys::JWKS, std::move(audiences));
-}
-
-void CreateAuthIssuers(
-    const HttpMixerConfig& config,
-    std::vector<std::shared_ptr<Auth::IssuerInfo>>* issuers) {
-  for (const auto& it : config.http_config.service_configs()) {
-    if (it.second.has_end_user_authn_spec()) {
-      for (const auto& jwt : it.second.end_user_authn_spec().jwts()) {
-        issuers->push_back(CreateIssuer(jwt));
-      }
-    }
+  issuer->name = jwt.issuer();
+  issuer->pubkey_type = Auth::Pubkeys::JWKS;
+  issuer->uri = jwt.jwks_uri();
+  issuer->cluster = jwt.jwks_uri_envoy_cluster();
+  if (jwt.has_public_key_cache_duration()) {
+    issuer->pubkey_cache_expiration_sec =
+        jwt.public_key_cache_duration().seconds();
   }
 }
 
 }  // namespace
 
-class Config {
+class Config : public Logger::Loggable<Logger::Id::http> {
```

Comment:
Can we have more descriptive name, and/or document what does this Config class mean?